File: 프로그래머스/0/181891. 순서 바꾸기/순서 바꾸기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int num_list[], size_t num_list_len, int n) {
    int* answer = (int*)malloc(num_list_len * sizeof(int));
    
    for(int i=0;i<num_list_len;i++){
        answer[i] = num_list[(i+n) % num_list_len];
    }
    
    return answer;
}

File: 프로그래머스/0/181891. 순서 바꾸기/순서 바꾸기.java
class Solution {
    public int[] solution(int[] num_list, int n) {
        int len = num_list.length;
        int[] answer = new int[len];
        
        for(int i=0;i<len;i++){
            answer[i] = num_list[(i+n) % len];
        }
        return answer;
    }
}